next-app/src/lib/forms/Button.svelte

```diff
@@ -3,7 +3,7 @@
 </script>
 
 <!-- https://daisyui.com/components/button -->
-<button on:click class:danger class='btn btn-primary'>
+<button on:click class:danger class='btn btn-primary { $$props.class }'>
```

Comment:
https://svelte.dev/docs#template-syntax-attributes-and-props recommends against using `$$props` for perfornance's sake (emphasis mine):

> `$$props` references all props that are passed to a component, including ones that are not declared with export. **It is not generally recommended**, as it is difficult for Svelte to optimise. But it can be useful in rare cases – for example, when you don't know at compile time what props might be passed to a component.

Is there any reason not to do `export let class: string = ''` in the component and then do `{ class }` here?

Reply:
I am aware of their advice on the use of `$$props` but have never found it to be an issue but am happy to incorporate a technique I've used in the past for this sort of thing.  Take a look and let me know what you think (https://github.com/sillsdev/web-languageforge/pull/1416/commits/18140fc6e8199d0a402c4be9aaa8da5b0390db2f).  Since `class` is a reserved word, we have to do some yucky workaround which is why I typically just stick with `$$props`.

Reply:
mind reviewing my last reply @rmunn ?

Reply:
Yes, the `export { className as class }` (or `classes` here, doesn't really matter what it's called) technique is nice, and I see that's what other people are recommending for this situation. I like it, and I feel okay with resolving this conversation. Sorry it lingered so long.

Reply:
see my other comment above in a different line.  I am fine with the resolution of keeping $$props.class, just recommending an explanatory comment to that effect.